src/ExpressionEvaluator/CSharp/Test/ResultProvider/FullNameTests.cs

```diff
@@ -909,5 +909,87 @@ public void BackTick_FirstCharacter()
             Verify(GetChildren(children.Single()),
                 EvalResult("x", "0", "int", fullName: null));
         }
+
+        [Fact]
+        public void MangledName_SimplifySynthesizedLocalName()
+        {
+            IDkmClrFullNameProvider2 fullNameProvider = new CSharpFormatter();
+            var inspectionContext = CreateDkmInspectionContext();
+            // The synthesized locals name should just become an empty string.
```

Comment:
>should just become an empty string [](http://example.com/codeflow?start=43&length=34)

What is the effect in the EE of returning an empty string for a synthesized local name? Where would this appear when debugging?

Reply:
This is used for the Null Reference Exception analysis where we get a identifier for something that was null. In such cases, we expect an empty string for the synthesized local name (the thing that will be null will be a field in it which is what we care about).
So the metadata name is actually `<CompilerGoop>SynthesizedLocalName.m_myUserLocalName was null` which becomes
`m_myUserLocalName was null` since the compiler returns empty for the synthesized local.